### ISSUE TITLE
[BUG] Many decimal integration test failures for 0.5

### ISSUE THREAD

**Issue by jlowe** [NVIDIA/spark-rapids#1802]
A number of unit tests failed on Databricks for the 0.5 nightly integration tests.  Some excerpts:
```
01:01:19  =========================== short test summary info ============================
01:01:19  FAILED ../../src/main/python/array_test.py::test_orderby_array_unique[Array(Decimal(7,-3))]
01:01:19  FAILED ../../src/main/python/array_test.py::test_orderby_array_unique[Array(Decimal(7,7))]
01:01:19  FAILED ../../src/main/python/array_test.py::test_orderby_array_unique[Array(Decimal(7,3))]
01:01:19  FAILED ../../src/main/python/array_test.py::test_orderby_array_unique[Array(Decimal(12,2))]
01:01:19  FAILED ../../src/main/python/array_test.py::test_orderby_array_unique[Array(Decimal(18,0))]
01:01:19  FAILED ../../src/main/python/cmp_test.py::test_dropna_any[Array(Decimal(7,-3))]
01:01:19  FAILED ../../src/main/python/cmp_test.py::test_dropna_any[Array(Decimal(18,0))]
01:01:19  FAILED ../../src/main/python/cmp_test.py::test_dropna_any[Array(Decimal(7,7))]
01:01:19  FAILED ../../src/main/python/cmp_test.py::test_dropna_all[Array(Decimal(7,-3))]
01:01:19  FAILED ../../src/main/python/cmp_test.py::test_dropna_any[Array(Decimal(7,3))]
01:01:19  FAILED ../../src/main/python/cmp_test.py::test_dropna_all[Array(Decimal(7,3))]
01:01:19  FAILED ../../src/main/python/cmp_test.py::test_dropna_any[Array(Decimal(12,2))]
01:01:19  FAILED ../../src/main/python/cmp_test.py::test_dropna_all[Array(Decimal(7,7))]
01:01:19  FAILED ../../src/main/python/cmp_test.py::test_dropna_all[Array(Decimal(18,0))]
01:01:19  FAILED ../../src/main/python/cmp_test.py::test_filter[Array(Decimal(7,7))] - ...
01:01:19  FAILED ../../src/main/python/cmp_test.py::test_dropna_all[Array(Decimal(12,2))]
01:01:19  FAILED ../../src/main/python/cmp_test.py::test_filter[Array(Decimal(7,3))] - ...
01:01:19  FAILED ../../src/main/python/cmp_test.py::test_filter[Array(Decimal(7,-3))]
01:01:19  FAILED ../../src/main/python/cmp_test.py::test_filter[Array(Decimal(18,0))]
01:01:19  FAILED ../../src/main/python/cmp_test.py::test_filter_with_project[Array(Decimal(7,-3))]
01:01:19  FAILED ../../src/main/python/cmp_test.py::test_filter_with_project[Array(Decimal(7,7))]
01:01:19  FAILED ../../src/main/python/cmp_test.py::test_filter[Array(Decimal(12,2))]
01:01:19  FAILED ../../src/main/python/cmp_test.py::test_filter_with_project[Array(Decimal(12,2))]
01:01:19  FAILED ../../src/main/python/cmp_test.py::test_filter_with_project[Array(Decimal(7,3))]
01:01:19  FAILED ../../src/main/python/cmp_test.py::test_filter_with_project[Array(Decimal(18,0))]
01:01:19  FAILED ../../src/main/python/window_function_test.py::test_window_aggs_for_rows_collect_list_no_nulls[IGNORE_ORDER({'local': True})]
```

**Comment by jlowe** [comment]
Detailed log of test errors:
[test_errors.txt](https://github.com/NVIDIA/spark-rapids/files/6036859/test_errors.txt)

**Comment by revans2** [comment]
All of these tests just failed for me in a pre-commit build for 0.5.  It feels almost like there is a change in cudf that is causing this I am going to try and track down what is happening.

**Comment by revans2** [comment]
Yup I was able to reproduce the failures on the bast 0.5 branch locally.

**Comment by revans2** [comment]
I went back a couple of days on the plugin and it still fails, this looks very much like a cudf issue.

**Comment by revans2** [comment]
Yup when I got back to 43b44e1ce447c52113b3abd87da026e9e7725ec9 in cudf all of the tests pass again. I will start to track down the change that caused this to happen

**Comment by jlowe** [comment]
This should be fixed when https://github.com/rapidsai/cudf/pull/7449 is merged and a new cudf-0.19 snapshot is published.

**Comment by revans2** [comment]
Dependency is merged and tests appear to be passing.
